src/borg/helpers/time.py

```diff
@@ -109,6 +110,48 @@ def format_timedelta(td):
     return txt
 
 
+def calculate_relative_offset(format_string, from_date, earlier=False):
+    """Calculates offset based on a relative marker. 7d (7 days), 8m (8 months)
+    earlier: whether offset should be calculated to an earlier time.
+    """
+    if from_date is None:
+        from_date = archive_ts_now().date()
+
+    if format_string is not None:
+        day_offset_regex = re.compile(r"(?P<offset>\d+)d")
+        month_offset_regex = re.compile(r"(?P<offset>\d+)m")
+
+        day_offset_match = day_offset_regex.search(format_string)
+        if day_offset_match:
+            day_offset = int(day_offset_match.group("offset"))
+            day_offset *= -1 if earlier else 1
+            return from_date + timedelta(days=day_offset)
+
+        month_offset_match = month_offset_regex.search(format_string)
+        if month_offset_match:
+            month_offset = int(month_offset_match.group("offset"))
+            month_offset *= -1 if earlier else 1
+            return offset_n_months(from_date, month_offset)
+
+    raise ValueError(f"{format_string} is not a recognized relative time marker")
+
+
+def offset_n_months(from_date, n_months):
+    # Calculate target month and year by getting completed total_months until target_month
+    total_months = (from_date.year * 12) + from_date.month + n_months
+    target_year = (total_months - 1) // 12
+    target_month = (total_months % 12) or 12
```

Comment:
guess that is correct, but i also guess easier to understand (div / mod == the pure arithmetic way, instead of the mixed arithmetic and boolean-switch way) would be:
```
target_month = (total_months - 1) % 12 + 1
```

Maybe even better / easier / more correct than that would be to move the "- 1" to the computation of total_months, so that it computes total_months == 0 for the very first month (year 0, month 1) and n_months == 0.